mlflow/pipelines/steps/split.py

```diff
@@ -226,9 +238,15 @@ def _run(self, output_directory):
             _logger.debug(
                 f"Running {post_split_filter_fn_name} on train, validation and test datasets."
             )
-            train_df = train_df[post_split_filter(train_df)]
```

Comment:
This is a great place to use decorators. You can keep the code intact but still perform output validation with a decorated `post_split_filter` method.